managed/services/telemetry/telemetry.go

```diff
@@ -160,7 +160,47 @@ func (s *Service) DistributionMethod() serverpb.DistributionMethod {
 	return s.sDistributionMethod
 }
 
-func (s *Service) prepareReport(ctx context.Context) *reporter.ReportRequest {
+func (s *Service) processSendCh(ctx context.Context) {
+	var inflightReports []*pmmv1.ServerMetric
+	var sendCtx context.Context
+	var cancel context.CancelFunc
+
+	for {
+		select {
+		case report, ok := <-s.sendCh:
+			if ok {
+				inflightReports = append(inflightReports, report)
+				if sendCtx != nil {
+					cancel()
+				}
+				sendCtx, cancel = context.WithCancel(context.Background())
```

Comment:
shouldn't it inherit context passed to `processSendCh`?

Reply:
yeah, make sense. thanks